### PR TITLE
feat: add support for Postgres multi dimensional arrays in Go

### PR DESCRIPTION
**Pull request by colli173**
This PR adds support for generating model fields from multi-dimensional postgresql array columns. The approach here is to pass around an additional column attribute `array_bounds` representing the dimensions of the postgres array column. When generating code, the `array_bounds` is checked along `is_array` to determine the dimensions of the resulting struct field. Would be happy to hear alternative approaches or solutions here as well. More info: [Postgres Arrays](https://www.postgresql.org/docs/14/arrays.html)

fixes issue: https://github.com/kyleconroy/sqlc/issues/1494

### CODE REVIEW

**Review by kyleconroy — DISMISSED** [review]
This looks like a solid improvement over our current multi-dimension support. The code generation looks correct, but I do want to verify that this works against a running PostgreSQL instance before merging.

Can you fix the merge conflicts? Happy to merge after that.

**Comment by colli173** [comment]
> We should only change the multi-dimension support for lib/pq.

Thanks for taking a look. Could you clarify the ask here? Is there an aspect of this that should change for the lib/pq package?

**Comment by colli173** [comment]
@kyleconroy soft ping for another review when you have a chance.

**Pull request closed without merging**
